sw/airborne/arch/chibios/mcu_periph/adc_arch.c

```diff
@@ -344,25 +349,45 @@ void adc_init(void)
   uint32_t smpr1, smpr2;
   adc_sample_time_on_all_channels(&smpr1, &smpr2, ADC_SAMPLE_RATE);
 
-  adcgrpcfg.cr2 = ADC_CR2_CFG;
-
 #if USE_ADC_WATCHDOG
   adc_watchdog.adc = NULL;
   adc_watchdog.cb = NULL;
   adc_watchdog.channel = 0;
-  adc_watchdog.vmin = (1<<12)-1; // max adc
+  adc_watchdog.vmin = (1 << 12) - 1; // max adc
 #endif
 
   adcgrpcfg.circular = TRUE;
   adcgrpcfg.num_channels = ADC_NUM_CHANNELS;
   adcgrpcfg.end_cb = adc1callback;
   adcgrpcfg.error_cb = adcerrorcallback;
+#if defined(__STM32F373xC_H)
```

Comment:
it would be nice to have a better way of specifying the driver than adding ifdefs, perhaps separate files for each chibios arch - but that is indeed out of scope of this PR